src/vcpkg/registries.cpp

```diff
@@ -1135,14 +1146,21 @@ namespace vcpkg
         return r.array_elements(arr, underlying);
     }
 
-    LockFile::Entry LockFile::get_or_fetch(const VcpkgPaths& paths, StringView key)
+    LockFile::Entry LockFile::get_or_fetch(const VcpkgPaths& paths, StringView repo, StringView reference)
     {
-        auto it = lockdata.find(key);
+        auto lockdata_key = repo.to_string();
+        if (reference != "HEAD")
+        {
+            // old-style, only repo. implicitly HEAD
+            Strings::append(lockdata_key, "@", reference);
+        }
```

Comment:
```suggestion
        auto lockdata_key = Strings::concat(repo, "@", reference);
```

We always want to use the `@` syntax from now on, and update old fields when we see them.